pkg/virtctl/expose/expose.go

```diff
@@ -0,0 +1,213 @@
+/*
+* This file is part of the KubeVirt project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Copyright 2018 Red Hat, Inc.
+*
+ */
+
+package expose
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/api/core/v1"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	COMMAND_EXPOSE = "expose"
+)
+
+type Command struct {
+	clientConfig clientcmd.ClientConfig
+	command      string
+}
+
+// holding flag information
+var serviceName string
+var clusterIP string
+var externalIP string
+var loadBalancerIP string
+var port int32
+var nodePort int32
+var strProtocol string
+var intTargetPort int
+var strServiceType string
+var portName string
+
+// generate a new "expose" command
+func NewExposeCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "expose TYPE NAME",
+		Short: "Expose a virtual machine as a new service.",
+		Long: `Looks up a virtual machine, offline virtual machine or virtual machine replica set by name and use its selector as the selector for a new service on the specified port.
+        A virtual machine replica set will be exposed as a service only if its selector is convertible to a selector that service supports, i.e. when the selector contains only the matchLabels component.
+        Note that if no port is specified via --port and the exposed resource has multiple ports, all will be re-used by the new service. 
+        Also if no labels are specified, the new service will re-use the labels from the resource it exposes.
+        
+        Possible types are (case insensitive, both single and plurant forms):
+        
+        virtualmachine (vm), offlinevirtualmachine (ovm), virtualmachinereplicaset (vmrs)`,
+		Example: usage(),
+		Args:    cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := Command{command: COMMAND_EXPOSE, clientConfig: clientConfig}
+			return c.RunE(cmd, args)
+		},
+	}
+
+	// flags for the "expose" command
+	cmd.Flags().StringVar(&serviceName, "name", "", "Name of the service created for the exposure of the VM")
```

Comment:
Could you distinguish this from positional arg NAME? Maybe service-name. Or change NAME to MACHINE_NAME

Reply:
i tried to follow "kubectl expose":
```
kubectl expose (-f FILENAME | TYPE NAME) [--port=port] [--protocol=TCP|UDP] [--target-port=number-or-name] [--name=name] [--external-ip=external-ip-of-service] [--type=type] [options]
```
they use "NAME" and "--name"